### PR TITLE
TF-2502 Display all day event

### PR DESCRIPTION
**Pull request by hieutbui**
## Issue
- #2502 

## Resolved

https://github.com/linagora/tmail-flutter/assets/80142234/6bafae5d-35e7-4ce6-a5fd-9948c2ea3f21

### CODE REVIEW

**Comment by dab246** [comment]
Please check condition `isAllDayEvent`

File: lib/features/email/presentation/extensions/calendar_event_extension.dart [hunk 1]
```diff
@@ -256,7 +256,37 @@ extension CalendarEventExtension on CalendarEvent {
...
+      final endMinute = endUtcDate!.value.minute;
+      log('CalendarEventExtension::endDate: $endUtcDate');
+      if (startHour == 0 && startMinute == 0 && endHour == 0 && endMinute == 0 && !DateUtils.isSameDay(localStartDate, localEndDate)) {
+        final timeZoneOffset = DateTime.now().timeZoneOffset.inHours;
+        final timeZoneOffsetAsString = timeZoneOffset >= 0 ? '+$timeZoneOffset' : '$timeZoneOffset';
+        final timeStart = formatDate(AppUtils.getCurrentDateLocale(), startUtcDate!.value);
+        final timeEnd = formatDate(AppUtils.getCurrentDateLocale(), endUtcDate!.value);
+        return '$timeStart - $timeEnd (GMT$timeZoneOffsetAsString)';
+      }
+    }
```
Comment (hoangdat) [inline lib/features/email/presentation/extensions/calendar_event_extension.dart:289]: hard for me to understand :(

File: lib/features/email/presentation/extensions/calendar_event_extension.dart [hunk 2]
```diff
@@ -256,7 +256,54 @@ extension CalendarEventExtension on CalendarEvent {
...
+
+    if (DateUtils.isSameDay(startDate, endDateToDisplay)) {
+      return '$dateStart (GMT$timeZoneOffsetAsString)';
+    } else {
+      return '$dateStart - $dateEnd (GMT$timeZoneOffsetAsString)';
+    }
+  }
+
   String get dateTimeEventAsString {
+    if (isAllDayEvent == true) {
```
Comment (dab246) [inline lib/features/email/presentation/extensions/calendar_event_extension.dart:303]:
```suggestion
    if (isAllDayEvent) {
```

Comment (hieutbui) [reply lib/features/email/presentation/extensions/calendar_event_extension.dart]: done

File: lib/features/email/presentation/extensions/calendar_event_extension.dart [hunk 3]
```diff
@@ -256,7 +256,60 @@ extension CalendarEventExtension on CalendarEvent {
...
+      final endDateValue = endUtcDate!.value;
+
+      final eventDurationInHours = startDateValue.difference(endDateValue).inHours;
+
+      final startHour = startDateValue.hour;
+      final startMinute = startDateValue.minute;
+      
+      final endHour = endDateValue.hour;
+      final endMinute = endDateValue.minute;
+      return startHour == 0 && startMinute == 0 && endHour == 0 && endMinute == 0 && eventDurationInHours % 24 == 0;
```
Comment (dab246) [inline lib/features/email/presentation/extensions/calendar_event_extension.dart:287]: IMO, we also need check `second` of start/end date

Comment (hieutbui) [reply lib/features/email/presentation/extensions/calendar_event_extension.dart]: done

**Review by dab246 — DISMISSED** [review]
- [x] PLease rebase

**Comment by hieutbui** [comment]
> * [ ]  PLease rebase

done

**Review by hoangdat — APPROVED** [review]
please rebase to `v0.11.3-patch4-dev`

**Comment by hieutbui** [comment]
> please rebase to `v0.11.3-patch4-dev`

done

**Review by hoangdat — APPROVED** [review]

**Pull request merged**